webapp/src/app/reducer.ts

```diff
@@ -12,13 +14,19 @@ export type Error = {
 };
 
 export type State = {
+  dogs: ILostDogWithPicture[] | any;
```

Comment:
not any, any means literally anything, if you mean nothing you have 'null' or 'undefined'

Reply:
eventually if it is empty array then, just "ILostDogWithPicture[]" without any other option